src/dodal/beamlines/b16.py

```diff
@@ -0,0 +1,65 @@
+from pathlib import Path
+
+from ophyd_async.epics.adcore import (
+    AreaDetector,
+)
+from ophyd_async.epics.motor import Motor
+
+from dodal.common.beamlines.beamline_utils import (
+    device_factory,
+    set_path_provider,
+)
+from dodal.common.beamlines.beamline_utils import set_beamline as set_utils_beamline
+from dodal.common.visit import RemoteDirectoryServiceClient, StaticVisitPathProvider
+from dodal.devices.b16.detector import (
+    software_triggered_tiff_area_detector,
+)
+from dodal.devices.motors import XYZPositioner
+from dodal.log import set_beamline as set_log_beamline
+from dodal.utils import BeamlinePrefix, get_beamline_name
+
+BL = get_beamline_name("b16")
+PREFIX = BeamlinePrefix(BL)
+set_log_beamline(BL)
+set_utils_beamline(BL)
+
+set_path_provider(
+    StaticVisitPathProvider(
+        BL,
+        Path("/dls/b16/data/2025/cm40635-3/bluesky"),
+        client=RemoteDirectoryServiceClient("http://b16-control:8088/api"),
+    )
+)
+
+
+@device_factory()
+def attol1() -> Motor:
+    return Motor("BL16B-EA-ECC-03:ACT0")
+
+
+@device_factory()
+def attol2() -> Motor:
+    return Motor("BL16B-EA-ECC-03:ACT1")
+
+
+@device_factory()
+def attol3() -> Motor:
+    return Motor("BL16B-EA-ECC-03:ACT2")
+
+
+@device_factory()
+def attorot1() -> Motor:
+    return Motor("BL16B-EA-ECC-02:ACT2")
```

Comment:
Could: I would pull these all into an `Attocube` device for neatness and then reference them in plans as e.g. `attocube.rot1`. I would also make the names more descriptive as someone who doesn't know the beamline it's not clear what the attocube actually moves or what axes `l1`, `l2` etc. are. My personal view is that in a world of autocomplete it's better to have longer descriptive names but even if you want short names a few docstrings on here that describe what the attocube moves and what each axis is would be great.

Reply:
Will have a think about how to organise these as b16 config grows. Although I would love to rename them (not really sure myself what the names mean) I'm hesitant to do that because staff have referred to these by these names for years. Similarly unsure about grouping them into devices - these four motors belong to two separate attocube controllers, and they have at least 6 controllers, and I don't think it means much to them which controller each axis belongs to. Maybe a generic MotorGroup device would be good for this, not necessarily mapping to a single physical device.

Reply:
> not really sure myself what the names mean

I think this definitely means that we should, if not rename them, then at least have good docstrings describing them.